Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+                    // if the array index doesn't exist, use the value for the highest index
+                    ValueType maxQualityLevel = GetHighestQualityForSetting(visitArgs.m_jsonKeyPath);
+                    if (maxQualityLevel < 0)
+                    {
+                        AZ_Warning("QualityCVarGroup", false, "Not applying settings for %.s because the array is empty.", AZ_STRING_ARG(command));
+                        return VisitResponse::Continue;
+                    }
+                    valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(visitArgs.m_jsonKeyPath), maxQualityLevel);
+                }
+            }
+
+            PerformConsoleCommand(command, valueKey);
+
+            return VisitResponse::Continue;
+        };
+        AZ::SettingsRegistryVisitorUtils::VisitObject(*m_settingsRegistry, settingsVisitorCallback, key);
+    }
+
+    void QualityCVarGroup::PerformConsoleCommand(AZStd::string_view command, AZStd::string_view key)
+    {
+        AZStd::string value;
+        switch (m_settingsRegistry->GetType(key))
+        {
+        case AZ::SettingsRegistryInterface::Type::FloatingPoint:
+            {
+                double floatingPointValue;
+                if (m_settingsRegistry->Get(floatingPointValue, key))
+                {
+                    value = AZStd::to_string(floatingPointValue);
+                }
+            }
+            break;
+        case AZ::SettingsRegistryInterface::Type::Boolean:
+            {
+                bool boolValue;
+                if (m_settingsRegistry->Get(boolValue, key))
+                {
+                    value = boolValue ? "True" : "False";
+                }
+            }
+            break;
+        case AZ::SettingsRegistryInterface::Type::Integer:
+            {
+                AZ::s64 intValue;
+                if (m_settingsRegistry->Get(intValue, key))
+                {
+                    value = AZStd::to_string(intValue);
+                }
+            }
+            break;
+        case AZ::SettingsRegistryInterface::Type::String:
+        default:
+            {
+                m_settingsRegistry->Get(value, key);
+            }
+            break;
+        }
+
+        if (!value.empty())
+        {
+            m_console->PerformCommand(command, { value }, AZ::ConsoleSilentMode::Silent);
+        }
+        else
+        {
+            AZ_Warning("QualityCVarGroup", false, "Failed to convert the value for %.*s to a string, or the value is empty.", AZ_STRING_ARG(command));
+        }
+    }
+
+    bool QualityCVarGroup::StringToValue(const AZ::ConsoleCommandContainer& arguments)
+    {
+        if (!arguments.empty())
+        {
+            AZ::CVarFixedString convertCandidate{ arguments.front() };
+            char* endPtr = nullptr;
+
+            // TODO support loading quality levels using names
+
+            ValueType qualityLevel = aznumeric_cast<ValueType>(strtoll(convertCandidate.c_str(), &endPtr, 0));

```

Comment:
For conversions to a value, this should use the FunctionObject of [AZ::ConsoleTypeHelpers::ToValue](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/ConsoleTypeHelpers.inl#L401-L419)